docs/architecture-decisions/0003-implement-as-a-graphql-api.md

```diff
@@ -0,0 +1,81 @@
+# 3. Implement as a GraphQL API
+
+Date: 2023-07-14
+
+## Status
+
+Accepted
+
+## Context
+
+In early design exercises, we envisioned TACOS more in terms of the staff workflow of analyzing search terms. In the
+longer term, we would also want the tool to be able to evaluate search terms algorithmically.
+
+In either case, the tool should be able to return its evaluation of a search term to a search interface, so that
+interface can better detect user intent. The basic flow might look like this:
+
+1. A user pastes a citation into the Bento input field and executes a search.
+2. Bento asks TACOS (possibly via ActiveRecord callbacks) to analyze the search term.
+3. TACOS evaluates the search term and hopefully identifies it as a citation.
+4. TACOS returns its evaluation to Bento.
+5. In the search results, Bento shares its best guess as to what the user was searching for. In this case, it would be
+a link to the article, but it might also be a database or journal, a library service offering, or contact information
+of a subject matter expert in the libraries.
+
+```mermaid
+flowchart LR
+  bento
+  tacosapi
+  analyze
+  returndata
+  ui
+
+  bento[1. Bento Search] -- API Request --> tacosapi[2. TACOS API] --> analyze[3. analyze with various algorithms] --> tacosapi 
+  
+  tacosapi -- API Response --> returndata[4. returnStucturedData] --> ui[5. Use data in UI]
+
+```
+
+This pattern would also replace the 'fact' feature built into the TIMDEX UI. In other words, TACOS would be responsible
+for matching DOIs, ISBNS, ISSNs, and other identifiers that could resolve known-item searches.
```

Comment:
I'm going to add a sentence here to clarify the responsibilities of TACOS vs UIs that use it: TIMDEX UI would be responsible for deciding what to do when TACOS returns structured data that includes information about a DOI, etc